### DISCUSSION TITLE
Variable reference error (export as ABQ files)

### DISCUSSION THREAD

**Discussion by elmeririzarry** [Concrete-Chrono-Development/chrono-preprocessor#17] [Q&A]
When attempting to export an LDPM mesh in ABAQUS format I encounter the following error (username redacted):

> 19:32:25  Running the Python command 'mod_LDPMCSL_gen' failed:
> Traceback (most recent call last):
>   File "C:\Users\-\AppData\Roaming\FreeCAD\Mod\chronoConcrete\freecad\chronoWorkbench\modules\mod_LDPMCSL_gen.py", line 522, in Activated
>     Gui.Control.showDialog(genWindow_LDPMCSL())
>   File "C:\Users\-\AppData\Roaming\FreeCAD\Mod\chronoConcrete\freecad\chronoWorkbench\modules\mod_LDPMCSL_gen.py", line 198, in __init__
>     modelInfo.append("Number of Nodes: " + str(numNodes))
> 
> local variable 'numNodes' referenced before assignment

At this step, the model mesh has been successfully generated and the only goal is to obtain all the files in ABAQUS format.

**Comment by mtroemner** [comment]
Hello, running a simple test of generating a mesh and then switching back to the Chrono workbench to generate an input file seemed to work fine in testing today. Could you please provide the .cwPar parameter file for the model you are generating? Thanks

**Comment by elmeririzarry** [comment]
Hello, here are the .cwPar and corresponding .vtk file used in the workbench. The mesh generation has been successful, I only had issues with exporting in ABQ format. Thanks.

[filesQA17.zip](https://github.com/user-attachments/files/18293333/filesQA17.zip)

**Comment by mtroemner** [comment]
The bug should now be fixed. There was an issue in naming not being properly applied to the imported data, which was automatically applied to internally generated meshed. Let us know if you have further issues.
